plugins/flatpak/gs-flatpak.c

```diff
@@ -3678,6 +3678,75 @@ app_has_local_source (GsApp *app)
 		(url != NULL && g_str_has_prefix (url, "file://"));
 }
 
+static guint64
+get_installation_dir_free_space (GsFlatpak *self)
+{
+	g_autoptr (GFile) installation_dir = NULL;
+	g_autoptr (GFileInfo) info = NULL;
+	g_autoptr (GError) local_error = NULL;
+	guint64 free_space = 0;
+
+	installation_dir = flatpak_installation_get_path (self->installation);
+
+	info = g_file_query_filesystem_info (installation_dir,
+					     G_FILE_ATTRIBUTE_FILESYSTEM_FREE,
+					     NULL,
+					     &local_error);
+	if (info == NULL)
+		g_warning ("Failed to query filesystem info: %s", local_error->message);
+
+	free_space = g_file_info_get_attribute_uint64 (info, G_FILE_ATTRIBUTE_FILESYSTEM_FREE);
+	return free_space;
+}
+
+static gboolean
+gs_flatpak_has_space_to_install (GsFlatpak *self, GsApp *app, GsAppList *list)
+{
+	guint64 free_space = 0;
+	guint64 space_required = 0;
+
+	space_required = gs_app_get_size_download (app);
+	if (space_required == GS_APP_SIZE_UNKNOWABLE)
+		g_warning ("Failed to query download size: %s", gs_app_get_unique_id (app));
+
+	space_required = space_required + (500 * 1024 * 1024); /* min-free-space-size = 500MB for Endless */
```

Comment:
Please declare a variable or a define with this value. So it's easier to spot, and by giving it a proper name, the code will be easier to read too. Something like
`#define EOS_MIN_FREE_SPACE_NEEDED_BYTES (500 * 1024 * 1024)`